docs: Update ckb genesis script list (#301)
nervosnetwork/rfcs: rfcs/0024-ckb-genesis-script-list/0024-ckb-genesis-script-list.md

---
Number: "0024"
Category: Informational
Status: Final
Author: Dylan Duan
Organization: Nervina Labs
Created: 2020-05-21
---

# CKB Genesis Script List

## Abstract

> Genesis scripts are the smart contracts built by the CKB team and set in the CKB genesis block. Genesis scripts provide core functions like (e.g. [SECP256k1/blake160](#secp256k1blake160) and [Nervos DAO](#nervos-dao)). This document presents the information of all Nervos CKB genesis scripts, including a brief introduction and _code_hash_, _hash_type_, _out_point_(_tx_hash_ and _index_), _dep_type_ on mainnet Lina and testnet Aggron.

Reviewer: > Genesis scripts are the smart contracts built by the CKB team and set in the CKB genesis block.

Strictly speaking `TYPE_ID` is a `builtin` script, and it is not set in the CKB genesis block, it is directly implemented in CKB (not run in CKB-VM) in Rust.

Reviewer 2: I suggest adding the above comment to `type id` section.

Author: > > Genesis scripts are the smart contracts built by the CKB team and set in the CKB genesis block.
> 
> Strictly speaking `TYPE_ID` is a `builtin` script, and it is not set in the CKB genesis block, it is directly implemented in CKB (not run in CKB-VM) in Rust.

I will add the comment to `Type ID` section.

> Genesis scripts are the smart contracts built and deployed by the CKB core team. Genesis scripts can provide core functions (e.g. [SECP256k1/blake160](#secp256k1blake160) and [Nervos DAO](#nervos-dao)), or other auxiliary infrastructure components. This document presents the information of all Nervos CKB genesis scripts, including a brief introduction and _code_hash_, _hash_type_, _out_point_(_tx_hash_ and _index_), _dep_type_ on mainnet Lina and testnet Aggron.

Reviewer 2: ```suggestion
Genesis scripts are the smart contracts built by the CKB team and set in the CKB genesis block. Genesis scripts provide core functions like (e.g. [SECP256k1/blake160](#secp256k1blake160) and [Nervos DAO](#nervos-dao)). This document presents the information of all Nervos CKB genesis scripts, including a brief introduction and _code_hash_, _hash_type_, _out_point_(_tx_hash_ and _index_), _dep_type_ on mainnet Lina and testnet Aggron.
```

| ----------- | -------------------------------------------------------------------- |
| `code_hash` | `0x82d76d1b75fe2fd9a27dfbaa65a039221a380d76c926f378d3f81cf3e7e13f2e` |
| `hash_type` | `type`                                                               |
| `tx_hash`   | `0xe2fb199810d49a4d8beec56718ba2593b665db9d52299a0f9e6e75416d73ff5c` |
| `index`     | `0x2`                                                                |
| `dep_type`  | `code`                                                               |

- Aggron

| parameter   | value                                                                |
| ----------- | -------------------------------------------------------------------- |
| `code_hash` | `0x82d76d1b75fe2fd9a27dfbaa65a039221a380d76c926f378d3f81cf3e7e13f2e` |
| `hash_type` | `type`                                                               |
| `tx_hash`   | `0x8f8c79eb6671709633fe6a46de93c0fedc9c1b8a6527a18d3983879542635c9f` |
| `index`     | `0x2`                                                                |
| `dep_type`  | `code`                                                               |

### Type ID

> [Type ID](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#type-id) ([Source Code](https://github.com/nervosnetwork/ckb/blob/master/script/src/type_id.rs)) is the builtin type script to support upgradable script. The _code_hash_ and _hash_type_ are fixed and the same on mainnet Lina and testnet Aggron and the `cell_type` is no need. 

Reviewer 2: ```suggestion
[Type ID](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#type-id) ([Source Code](https://github.com/nervosnetwork/ckb/blob/master/script/src/type_id.rs)) is the builtin type script to support unique and immutable reference. The _code_hash_ and _hash_type_ are fixed and the same on mainnet Lina and testnet Aggron and the `dep_type` is no need. 
```

> [Type ID](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#type-id) ([Source Code](https://github.com/nervosnetwork/ckb/blob/master/script/src/type_id.rs)) is the builtin type script, not set in the CKB genesis block, but directly implemented in CKB (not run in CKB-VM) in Rust. Type ID is usually used to support unique and immutable reference. The _code_hash_ and _hash_type_ are fixed and the same on mainnet Lina and testnet Aggron and the `dep_type` is no need. 

Reviewer 2: ```suggestion
Strictly speaking, [Type ID](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#type-id) ([Source Code](https://github.com/nervosnetwork/ckb/blob/master/script/src/type_id.rs)) is a builtin script directly implemented in Rust within CKB, it doesn't run in CKB-VM. It's still included in this document because it can be used the same way as other genesis scripts since genesis block.

Type ID is used to support unique and immutable references. The _code_hash_ and _hash_type_ are fixed and the same on mainnet Lina and testnet Aggron and the `dep_type` is no need. 
```
